sql/thriftserver/if/TCLIService.thrift

```diff
@@ -0,0 +1,1298 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Coding Conventions for this file:
+//
+// Structs/Enums/Unions
+// * Struct, Enum, and Union names begin with a "T",
+//   and use a capital letter for each new word, with no underscores.
+// * All fields should be declared as either optional or required.
+//
+// Functions
+// * Function names start with a capital letter and have a capital letter for
+//   each new word, with no underscores.
+// * Each function should take exactly one parameter, named TFunctionNameReq,
+//   and should return either void or TFunctionNameResp. This convention allows
+//   incremental updates.
+//
+// Services
+// * Service names begin with the letter "T", use a capital letter for each
+//   new word (with no underscores), and end with the word "Service".
+
+namespace java org.apache.spark.service.rpc.thrift
+namespace cpp apache.spark.service.rpc.thrift
```

Comment:
The org.apache.hive.service package was coming because it was in the hive-service module in Hive.
Since now it's all in one spark-thriftserver module, maybe change the package to `org.apache.spark.sql.thriftserver.cli.thrift` (as I suppose the other code will be in `org.apache.spark.sql.thriftserver`)

Reply:
> thrift

In hive-2.3.x, thrift protocol code have been separated from package name `org.apache.hive.service.cli.thrift`, now is `org.apache.spark.service.rpc.thrift`. 
because under package `org.apache.spark.service.rpc.thrift`, there are some other classes based on these generated thrift code.

Reply:
> The org.apache.hive.service package was coming because it was in the hive-service module in Hive.
> Since now it's all in one spark-thriftserver module, maybe change the package to `org.apache.spark.sql.thriftserver.cli.thrift` (as I suppose the other code will be in `org.apache.spark.sql.thriftserver`)


In hive-2.3.x, thrift protocol code have been separated from package name `org.apache.hive.service.cli.thrift`, now is `org.apache.spark.service.rpc.thrift`. 
because under package `org.apache.spark.service.rpc.thrift`, there are some other classes based on these generated thrift code.

Reply:
Does `hive-jdbc` still support it if we rename the package to `org.apache.spark.service.rpc.thrift`?

Reply:
@wangyum `hive-jdbc` will then not pickup this code, but instead use the code from the `hive-service` dependency.
One may say that it is bad, because it makes the client and server run different code, but I'd say is actually good - most people would use a standalone hive jdbc client jar that uses `hive-service` code from Hive. When we use it here, we test that the code in Spark does not break compatibility with Hive client, even if in the future we decide to make some changes to it.

Reply:
> @wangyum `hive-jdbc` will then not pickup this code, but instead use the code from the `hive-service` dependency.
> One may say that it is bad, because it makes the client and server run different code, but I'd say is actually good - most people would use a standalone hive jdbc client jar that uses `hive-service` code from Hive. When we use it here, we test that the code in Spark does not break compatibility with Hive client, even if in the future we decide to make some changes to it.

Yea, it is feasible if it meets the standards of the protocol and is backwards compatible. We are use a server implement  thriftserver based on v9, and it works well.

Reply:
> In hive-2.3.x, thrift protocol code have been separated from package name org.apache.hive.service.cli.thrift, now is org.apache.spark.service.rpc.thrift.
because under package org.apache.spark.service.rpc.thrift, there are some other classes based on these generated thrift code.

Could we use `org.apache.spark.thriftserver` everywhere instead of `org.apache.spark.service`? the `org.apache.hive.service` was coming from that `hive-service` was a separate module, but we don't have it here.

Reply:
> Could we use `org.apache.spark.thriftserver` everywhere instead of `org.apache.spark.service`? the `org.apache.hive.service` was coming from that `hive-service` was a separate module, but we don't have it here.

It is easy after  https://github.com/apache/spark/pull/26221#discussion_r338677057
It is ok to do like this since our package is `thriftserver` not `service`